rows.go

```diff
@@ -127,6 +129,103 @@ type Rows struct {
 	closeErr  error
 }
 
+// new Rows and add a set of driver.Value by using the struct reflect tag
+func newRowsFromStruct(m interface{}, tagName ...string) (*Rows, error) {
+	val := reflect.ValueOf(m).Elem()
+	num := val.NumField()
+	if num == 0 {
+		return nil, errors.New("no properties available")
+	}
+	columns := make([]string, 0, num)
+	var values []driver.Value
+	tag := "json"
+	if len(tagName) > 0 {
+		tag = tagName[0]
+	}
+	for i := 0; i < num; i++ {
+		f := val.Type().Field(i)
+		column := f.Tag.Get(tag)
+		if len(column) > 0 {
+			columns = append(columns, column)
+			values = append(values, val.Field(i).Interface())
+		}
+	}
+	if len(columns) == 0 {
+		return nil, errors.New("tag not match")
+	}
+	rows := &Rows{
+		cols:      columns,
+		nextErr:   make(map[int]error),
+		converter: driver.DefaultParameterConverter,
+	}
+	return rows.AddRow(values...), nil
+}
+
+// NewRowsFromInterface new Rows from struct or slice or array reflect with tagName
+// NOTE: arr/slice must be of the same type
+// tagName default "json"
+func NewRowsFromInterface(m interface{}, tagName string) (*Rows, error) {
```

Comment:
Go has generics now. It would be better to have multiple functions with stronger typing.

NewRowsFromSlice
NewRowsFromStruct